src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -35,21 +63,73 @@ public LineResponse saveLine(LineRequest request) {
     @Transactional(readOnly = true)
     public LineResponse findLineById(long id) {
         Optional<Line> line = lineRepository.findOneById(id);
-
-        return line.map(LineResponse::of).orElse(null);
+        return line.map(LineResponse::of).orElseThrow(() -> new ApplicationException(ApplicationType.CONTENT_NOT_FOUND));
     }
 
     public void deleteLineById(Long id) {
-        lineRepository.deleteById(id);
+        Optional<Line> line = lineRepository.findOneById(id);
+        line.orElseThrow(() -> new ApplicationException(ApplicationType.CONTENT_NOT_FOUND));
+
+        lineRepository.delete(line.get());
     }
 
     public LineResponse updateLine(Long id, LineRequest lineRequest) {
+        Optional<Line> persistLine = lineRepository.findOneById(id);
+
+        Line line = persistLine.orElseThrow(() -> new ApplicationException(ApplicationType.CONTENT_NOT_FOUND));
+        line.update(lineRequest.toLine());
+        return  LineResponse.of(line);
+    }
+
+    //구간 제거
+    /*
+    지하철 노선에 구간을 제거하는 기능 구현
+    지하철 노선에 등록된 마지막 역(하행 종점역)만 제거할 수 있다.
+    지하철 노선에 상행 종점역과 하행 종점역만 있는 경우(구간이 1개인 경우) 역을 삭제할 수 없다.
+    새로운 구간 제거시 위 조건에 부합하지 않는 경우 에러 처리한다.
+    */
+
+    //등록된 구간 통해 역 목록 조회
+    /*
+    지하철 노선 조회 시 등록된 역 목록을 함께 응답
+    노선에 등록된 구간을 순서대로 정렬하여 상행 종점부터 하행 종점까지 목록을 응답하기
+    * */
+
+    private void validateLineRequest(LineRequest lineRequest) {
+        lineRequest.validate();
+        stationService.validateStationId(lineRequest.getDownStationId());
+    }
+
+    public LineResponse createLineSection(Long id, SectionRequest sectionRequest) {
+        Optional<Line> line = lineRepository.findOneById(id);
+
+        line.orElseThrow(() -> new ApplicationException(ApplicationType.INVALID_ID));
```

Comment:
Optional에서 계속 값을 빼오기보다는 (113, 117, 120 line) 한줄로 처리해도 되지 않을까 싶습니다!
```suggestion
        Line line = lineRepository.findOneById(id)
            .orElseThrow(() -> new ApplicationException(ApplicationType.INVALID_ID));
```